src/scripts/mentoring-data-migration.js

```diff
@@ -0,0 +1,1325 @@
+						AND (tenant_code IS NULL OR tenant_code = '' OR organization_code IS NULL OR organization_code = '')
+					`,
+						{ transaction }
+					)
+
+					totalUpdated += metadata.rowCount || 0
+				}
+			}
+
+			// Clean up temp table
+			await this.sequelize.query(`DROP TABLE IF EXISTS temp_user_org_lookup`, { transaction })
+
+			console.log(
+				`✅ Updated ${totalUpdated} user_extensions records using GROUP BY organization_id approach (${Math.ceil(
+					orgIds.length / orgBatchSize
+				)} org batches)`
+			)
+			this.stats.successfulUpdates += totalUpdated
+
+			await transaction.commit()
+		} catch (error) {
+			await transaction.rollback()
+			throw error
+		}
+	}
+
+	/**
+	 * Process table with user_id using user_extensions updated data
+	 */
+	async processTableWithUserId(tableConfig) {
+		const { name, updateColumns, hasPartitionKey } = tableConfig
+		console.log(`\n🔄 Processing table with user_id: ${name}`)
+
+		try {
+			// Check if table exists and has target columns
+			const existingColumns = await this.checkTableColumns(name)
+			const availableUpdateColumns = updateColumns.filter((col) => existingColumns.includes(col))
+
+			if (availableUpdateColumns.length === 0) {
+				console.log(`⚠️  Table ${name} has no target columns, skipping`)
+				return
+			}
+
+			// Only update tenant_code for user_id tables
+			if (availableUpdateColumns.includes('tenant_code')) {
+				const citusEnabled = await this.isCitusEnabled()
+
+				// Undistribute if needed
+				let wasDistributed = false
+				if (citusEnabled && hasPartitionKey) {
+					wasDistributed = await this.undistributeTable(name)
+				}
+
+				try {
+					await this.executeUserIdUpdates(name)
+
+					// Add detailed logging like in migration
+					await this.logTableUpdateResults(name)
+
+					// Redistribute if needed
+					if (citusEnabled && wasDistributed) {
+						await this.redistributeTable(name)
+					}
+				} catch (error) {
+					console.error(`❌ Error during updates for ${name}:`, error.message)
+					throw error
+				}
+			}
+
+			console.log(`✅ Completed ${name}`)
+		} catch (error) {
+			console.error(`❌ Error processing table ${name}:`, error)
+			throw error
+		}
+	}
+
+	/**
+	 * Execute updates for tables with user_id using user_extensions data
+	 */
+	async executeUserIdUpdates(tableName) {
+		console.log(`📊 Executing user_id updates for ${tableName}...`)
+
+		const transaction = await this.sequelize.transaction()
+
+		try {
+			// Get table configuration
+			const tableConfig = this.tablesWithUserId.find((t) => t.name === tableName)
+
+			let updateQuery
+
+			if (tableConfig.useSessionLookup) {
+				// Special case for post_session_details: get data from sessions table based on session_id
+				const sessionIdColumn = tableConfig.columns.session_id
+
+				// Build SET clause based on available update columns
+				const setClauses = []
+				if (tableConfig.updateColumns.includes('tenant_code')) {
+					setClauses.push('tenant_code = s.tenant_code')
```

Comment:
@sumanvpacewisdom you are reffering the sessions table for getting the tenent code, you should take from csv